File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int n = s.size();
        int resIndex =0,resLength =0;
        vector<vector<bool>> dp(n,vector<bool>(n,false));

        for(int i = n-1;i>=0 ;i --){
            for(int j = i;j<n;j++){
                if(s[i] == s[j] && (j - i +1 <= 2 || dp[i+1][j-1])){
                    dp[i][j] = true;
                    
                }
            }
        }

        int count = 0;
        for(int i =0;i<n;i++){
            for(int j = 0; j <n ;j++){
                if(dp[i][j]) count ++;
            }
        }
        return count;
    }
};

File: Data Structures & Algorithms/last-stone-weight/submission-0.cs
public class Solution {
    public int LastStoneWeight(int[] stones) {
        PriorityQueue<int,int> maxHeap = new PriorityQueue<int,int>(
            Comparer<int>.Create((x,y)=>y.CompareTo(x))
        );
        foreach(int stone in stones){
            maxHeap.Enqueue(stone,stone);
        }

        while(maxHeap.Count >1){
            int x = maxHeap.Dequeue();
            int y = maxHeap.Dequeue();

            if(x < y){
                y -=x;
                maxHeap.Enqueue(y,y);
            }else if(y<x){
                x -= y;
                maxHeap.Enqueue(x,x);
            }
        }
        if(maxHeap.Count == 1){
            return maxHeap.Peek();
        }
        return 0;
    }
}
